### PR TITLE
PE-71 | Created Route Names for Screens

### PR DESCRIPTION
**Pull request by ayeejane**
**What I Did:** 

- Created a service file named `route_generator` that takes care of the navigation with screens without parameters. 
- Changed some filenames from page to screen to keep our file names uniform. 
- Removed unused imports. 

**How to Test:** 

- Uninstall the app from your emulator or physical phone. 
- Navigate through the app and make sure everything still works. Make sure to click on every button.

### CODE REVIEW

File: lib/screens/signin_screen.dart [hunk 1]
```diff
@@ -74,13 +72,7 @@ class _LoginPageState extends State<LoginPage> {
...
       await _emailAuth.signIn(email, password);
 
-      Navigator.pushAndRemoveUntil(
-        context,
-        MaterialPageRoute(
-          builder: (context) => Home(),
-        ),
-        (Route<void> route) => false,
-      );
+      Navigator.of(context).pushNamed('/home');
```
Comment (MaxLian11) [inline lib/screens/signin_screen.dart]: Is it possible to keep `(Route<void> route) => false,` as we don't want the user to go back to sign-in or profile onboarding screen page with the back button?

File: lib/screens/profile_onboarding_screen.dart [hunk 2]
```diff
@@ -205,12 +204,8 @@ class _ProfileOnboardingState extends State<ProfileOnboarding> {
...
         await _profileOnboardingService.addUser(user);
 
-        Navigator.pushAndRemoveUntil(
-          context,
-          MaterialPageRoute(
-            builder: (context) => Home(),
-          ),
-          (Route<void> route) => false,
+        Navigator.of(context).pushNamed(
+          '/home',
```
Comment (MaxLian11) [inline lib/screens/profile_onboarding_screen.dart]: Same here

File: lib/screens/settings/account/account_settings_delete.dart [hunk 3]
```diff
@@ -39,11 +38,7 @@ class _AccountSettingsDeleteState extends State<AccountSettingsDelete> {
       }
       await DeleteUserService().deleteUser(tempId);
 
-      Navigator.pushAndRemoveUntil(
-          context,
-          MaterialPageRoute<Widget>(
-              builder: (BuildContext context) => LoginPage()),
-          (Route<void> route) => false);
+      Navigator.of(context).pushNamed('/signIn');
```
Comment (MaxLian11) [inline lib/screens/settings/account/account_settings_delete.dart]: Same here

File: lib/screens/settings_screen.dart [hunk 4]
```diff
@@ -32,7 +32,7 @@ class SettingsScreen extends StatelessWidget {
     _emailAuth.signOut().then((_) => Navigator.pushAndRemoveUntil(
         context,
         MaterialPageRoute<Widget>(
-            builder: (BuildContext context) => RootPage()),
+            builder: (BuildContext context) => RootScreen()),
         (Route<void> route) => false));
```
Comment (MaxLian11) [inline lib/screens/settings_screen.dart]: This wasn't changed to the new route generator, was there a reason why? Or maybe I do  not completely understand the purpose of it

**Review by MaxLian11 — CHANGES_REQUESTED** [review]
Looks great, however, I believe we need to keep the cases in which User cannot go back after going to a new route, like in case with SignIn -> HomeScreen

**Review by tonynguyen98 — CHANGES_REQUESTED** [review]
I agree with Max's comments, it can be a security issue to allow users to go back on certain pages.

I believe you can do that with something like this: 
```dart
Navigator.pushNamedAndRemoveUntil(context, '/{routeName}', (Route<void> route) => false));
```

File: lib/screens/settings_screen.dart (hunk 4 above)
Comment (tonynguyen98) [reply lib/screens/settings_screen.dart]: Seems like you've missed this route?

**Review by dominikyz — APPROVED** [review]
If all of the requested changes are fixed, then we good to go. Good job! :)

**Review by tonynguyen98 — CHANGES_REQUESTED** [review]
Sign in with Google seems to be broken. When I press the button to sign in, nothing happens.

**Comment by tonynguyen98** [comment]
> Sign in with Google seems to be broken. When I press the button to sign in, nothing happens.

Apologies, it turned out that my emulator was bugging out. Everything is working as intended.

**Review by tonynguyen98 — APPROVED** [review]

**Review by MaxLian11 — APPROVED** [review]

**Pull request merged**